src/background/services/monetization.ts

```diff
@@ -55,9 +54,17 @@ export class MonetizationService {
       this.sessions[tabId] = new Map()
     }
 
+    const sessionsCount = this.sessions[tabId].size + payload.length
```

Comment:
Question: shouldn't we exclude `remove` intents from `payload.length` here?
Also, do we emit stopPaymentSession before startPaymentSession so existing sessions get removed first?